libraries/core_libs/consensus/src/transaction/gas_pricer.cpp

```diff
@@ -0,0 +1,91 @@
+#include "transaction/gas_pricer.hpp"
+
+namespace taraxa {
+
+GasPricer::GasPricer(uint64_t percentile, uint64_t number_of_blocks, std::shared_ptr<DbStorage> db)
+    : percentile_(percentile), price_list_(number_of_blocks) {
+  if (db) {
+    init_daemon_ = std::make_unique<std::thread>([this, db_ = std::move(db)]() { init(std::move(db_)); });
+  }
+}
+
+GasPricer::~GasPricer() {
+  if (init_daemon_) init_daemon_->join();
```

Comment:
if (init_daemon_ && init_daemon_->joinable())

Reply:
? this will not compile

Reply:
this only applies if you want to join current thread (that's not joinable), that's in this case impossible

Reply:
not true, thread is not joinable also if you moved it, detached or joined before, which is possible in case someone else changes the code and forgets about this